### PR TITLE
Improve test coverage for HooksConfigLib

### LINKED ISSUE #717: Coverage Report - Test HooksConfigLib.sol
Test the following functions with the described scenarios:

- [x] callBeforeAddLiquidityHook (hook returns false)
- [x] callAfterAddLiquidityHook (hook returns false)
- [x] callBeforeRemoveLiquidityHook (hook returns false)
- [x] callAfterRemoveLiquidityHook (hook returns false)
- [x] callAfterAddLiquidityHook (hook returns empty hookAdjustedAmounts)
- [x] callAfterRemoveLiquidityHook (hook returns empty hookAdjustedAmounts)

### PR DESCRIPTION
**Pull request by joaobrunoah**
# Description

The following tests were added:

- callBeforeAddLiquidityHook (hook returns false)
- callAfterAddLiquidityHook (hook returns false)
- callBeforeRemoveLiquidityHook (hook returns false)
- callAfterRemoveLiquidityHook (hook returns false)
- callAfterAddLiquidityHook (hook returns empty hookAdjustedAmounts)
- callAfterRemoveLiquidityHook (hook returns empty hookAdjustedAmounts)

It increased line coverage from 90% to 95%, but branch coverage is still 41% for this file. Still, it seems to be related to forge, I'll leave it be for now.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [x] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

## Issue Resolution

Closes #717

### CODE REVIEW

**Review by elshan-eth — APPROVED** [review]
Cool!

File: pkg/vault/contracts/lib/HooksConfigLib.sol [hunk 1]
```diff
@@ -332,7 +332,7 @@ library HooksConfigLib {
             params.userData
         );
 
-        if (success == false) {
+        if (success == false || hookAdjustedAmountsInRaw.length != amountsInRaw.length) {
```
Comment (jubeira) [inline pkg/vault/contracts/lib/HooksConfigLib.sol:335]: huh nice catch

File: pkg/vault/test/.contract-sizes/Vault [hunk 2]
```diff
@@ -1,2 +1,2 @@
-Bytecode	23.968
-InitCode	25.250
\ No newline at end of file
+Bytecode	24.005*
```
Comment (jubeira) [inline pkg/vault/test/.contract-sizes/Vault:1]: :sob:

**Review by jubeira — APPROVED** [review]
LGTM, thanks @joaobrunoah ! Great job.

File: pkg/vault/test/.contract-sizes/Vault (hunk 2 above)
Comment (joaobrunoah) [reply pkg/vault/test/.contract-sizes/Vault:1]: Sorry hahahahaha

**Pull request merged**